src/pybind/mgr/dashboard/services/nvmeof_conf.py

```diff
@@ -82,16 +92,33 @@ def get_service_info(cls, group=None):
             if not gateways:
                 return None
 
-            if group:
-                for service_name, entries in gateways.items():
-                    if group in service_name:
-                        entry = next((entry for entry in entries if entry['group'] == group), None)
-                        if entry['group'] == group:  # type: ignore
-                            return service_name, entry['service_url']  # type: ignore
+            try:
+                if group:
+                    orch = OrchClient.instance()
+
+                    for service_name, entries in gateways.items():
+                        # nvmeof naming convention is nvmeof.<rbd-pool>.<group>
```

Comment:
Thanks! Much appreciated.

Reply:
That said... Can there exist `nvmeof.rbd-1.group1` and `nvmeof.rbd-2.group1`? The group name should allow to univocally find the Cephadm service... I find weird having to iterate through nvmeof services, because we are not given all the info required...

Reply:
thanks. I removed this logic and instead I am getting the group name from the service spec itself. but 

>Can there exist nvmeof.rbd-1.group1 and nvmeof.rbd-2.group1? 

 looks like a valid question. @adk3798 is there any validation there for disallowing duplicate names? I guess logically people would want to chose different group names for each services unless there is some scenario which is supported by nvmeof

Reply:
@caroav can you also please have a look at this question?

> Can there exist `nvmeof.rbd-1.group1` and `nvmeof.rbd-2.group1` services?

Exposing the pool name complicates mapping group names to service names (we cannot infer the service name from `nvmeof.<group_name>`, but have to iterate through the `nvmeof` service list).

Reply:
Currently there is no enforcement on these rules. This is not good but that's where we are. The best practice will be to use a single pool, and within that pool, define the groups. I will discuss with cephadm how to add validations but it will not be for 8.0.

Reply:
@caroav it's no big issue, but if there's no benefit from it, changing that would simplify both code and operations. There would be a 1:1 mapping between a GW group and a Cephadm service.

Otherwise, we will also need to document the look-up process (e.g.: find a Cephadm service whose end part matches the GW group name).

@adk3798 why it was decided to include the pool name in the Cephadm service name? I assume that the ceph-nvmeof pool will also be defined in the spec, won't it?

Reply:
@epuertat I take the responsibility for not defining it properly. I will work with Adam to set the rules for next versions.

Reply:
The initial implementation had no concept of a group and we were already using pools for the default name for iscsi so we went with that. It's also worth noting it's possible to define the `service_id` to whatever you want in a spec file so even regardless of whether the default is `<pool>` or `<pool>.<group>` the name isn't a fully reliable way to get information about the service.

Reply:
> thanks. I removed this logic and instead I am getting the group name from the service spec itself. but
> 
> > Can there exist nvmeof.rbd-1.group1 and nvmeof.rbd-2.group1?
> 
> looks like a valid question. @adk3798 is there any validation there for disallowing duplicate names? I guess logically people would want to chose different group names for each services unless there is some scenario which is supported by nvmeof

if another service is created with the same service name it fully replaces the service whose name it matches (unless --no-overwrite is passed to the apply command)